packages/governance/test/swingsetTests/contractGovernor/governParam.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+
+import { Far } from '@agoric/marshal';
+import { E } from '@agoric/eventual-send';
+import { q, details as X } from '@agoric/assert';
+
+import { ChoiceMethod } from '../../../src/ballotBuilder';
+
+const setupGovernance = (paramManager, registrar) => {
+  const params = E(paramManager).getParams();
+  const validNames = Object.keys(params);
+
+  /**
+   * @param {string} name
+   * @param {ParamValue} proposedValue
+   * @param {ClosingRule} closingRule
+   */
+  const setParamByVoting = async (name, proposedValue, closingRule) => {
+    assert(
+      validNames.includes(name),
+      X`Name ${name} was not a governable parameter`,
+    );
+    /** @type {ParamDescription["type"]} */
+    // TODO: use paramDescription.type to actually enforce the type
+    // const paramType = params[name].type;
+
+    const positivePosition = {
```

Comment:
I think the positions and questions need to be records, maybe with text somehow indicated. This should be informed by how the wallet UI would display voting, which should replace presences with the user's petname for that presence.